packages/react/plugins/component-testing/index.ts

```diff
@@ -158,39 +185,68 @@ function buildTargetWebpack(
       : false;
   return getWebConfig(
     workspaceRoot,
-    thisProjectConfig.root,
-    thisProjectConfig.sourceRoot,
+    ctProjectConfig.root,
+    ctProjectConfig.sourceRoot,
     options,
     true,
     isScriptOptimizeOn,
-    configuration
+    parsed.configuration
   );
 }
 
 function getConfigByPath(
   graph: ProjectGraph,
   configPath: string
 ): ProjectConfiguration {
-  const configFileFromWorkspaceRoot = relative(
-    workspaceRoot,
-    configPath
-  ).replace(extname(configPath), '');
+  const configFileFromWorkspaceRoot = relative(workspaceRoot, configPath);
+  const normalizedPathFromWorkspaceRoot = lstatSync(configPath).isFile()
+    ? configFileFromWorkspaceRoot.replace(extname(configPath), '')
+    : configFileFromWorkspaceRoot;
+
   const mappedGraph = mapProjectGraphFiles(graph);
   const componentTestingProjectName =
-    mappedGraph.allFiles[configFileFromWorkspaceRoot];
+    mappedGraph.allFiles[normalizedPathFromWorkspaceRoot];
   if (
     !componentTestingProjectName ||
     !graph.nodes[componentTestingProjectName]?.data
   ) {
     throw new Error(
-      stripIndents`Unable to find the project configuration that includes ${configFileFromWorkspaceRoot}. 
+      stripIndents`Unable to find the project configuration that includes ${normalizedPathFromWorkspaceRoot}. 
       Found project name? ${componentTestingProjectName}. 
       Graph has data? ${!!graph.nodes[componentTestingProjectName]?.data}`
     );
   }
   // make sure name is set since it can be undefined
-  graph.nodes[componentTestingProjectName].data.name =
-    graph.nodes[componentTestingProjectName].data.name ||
+  graph.nodes[componentTestingProjectName].data.name ??=
     componentTestingProjectName;
   return graph.nodes[componentTestingProjectName].data;
 }
+
+function createExecutorContext(
+  graph: ProjectGraph,
+  targets: Record<string, TargetConfiguration>,
+  projectName: string,
+  targetName: string,
+  configurationName: string
+) {
+  const projectConfigs: { [projectNames: string]: ProjectConfiguration } =
+    Object.entries(graph.nodes).reduce((acc, [key, value]) => {
+      acc[key] = value.data;
+      return acc;
+    }, {});
```

Comment:
Use `readProjectsConfigurationFromProjectGraph` in `packages/nx/src/project-graph/project-graph.ts`